smack-extensions/src/main/java/org/jivesoftware/smackx/softwareinfo/SoftwareInfoManager.java

```diff
@@ -0,0 +1,143 @@
+/**
+ *
+ * Copyright 2019 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.softwareinfo;
+
+import java.io.IOException;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.WeakHashMap;
+
+import org.jivesoftware.smack.Manager;
+import org.jivesoftware.smack.SmackException.NoResponseException;
+import org.jivesoftware.smack.SmackException.NotConnectedException;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.XMPPException.XMPPErrorException;
+import org.jivesoftware.smack.parsing.SmackParsingException;
+import org.jivesoftware.smack.xml.XmlPullParserException;
+import org.jivesoftware.smackx.disco.ServiceDiscoveryManager;
+import org.jivesoftware.smackx.disco.packet.DiscoverInfo;
+import org.jivesoftware.smackx.disco.packet.DiscoverInfo.Identity;
+import org.jivesoftware.smackx.disco.packet.DiscoverInfoBuilder;
+import org.jivesoftware.smackx.softwareinfo.element.SoftwareInfoElement;
+
+import org.jxmpp.jid.Jid;
+
+/**
+* Entry point for Smack's API for XEP-0232: Software Information.
+* <br>
+* In order to be updated with other user's Software Information add {@link SoftwareInfoListener}, use {@link #addSoftwareInfoListener(SoftwareInfoListener)}.
+* <br>
+* @see <a href="https://xmpp.org/extensions/xep-0232.html">
+*     XEP-0232 : Software Information.
+*     </a>
+*/
+public final class SoftwareInfoManager extends Manager {
```

Comment:
What this manager really needs to provide is not `(add|remove)SoftwareListener()` API, but one that takes a `SoftwareInfoForm` converts it into a `DataForm` and injects it into the disco#info reply using `ServiceDiscoveryManager.setExtendedInfo(DataForm)`.